src/context/ProfileContext.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from 'react'
+
+const ProfileContext = React.createContext({
+   userProfile: null,
+   setUserProfile: () => {},
+   unsetUserProfile: () => {},
+})
+
+export const ProfileContextProvider = (props) => {
+   const [profile, setProfile] = useState(null)
+
+   const setUserProfile = (profile) => {
+      setProfile(profile)
+   }
+
+   const unsetUserProfile = () => {
+      setProfile(null)
+   }
+
+   useEffect(() => {
+      try {
+         const existingProfile = JSON.parse(localStorage.getItem('profile'))
```

Comment:
Instead of saving and reading the profile from Local Storage, let's call the API to get the profile data each time the app is loaded, a few reasons for this 
- We don't have to optimize API calls so early, it's ok to have API calls :)
- When the user changes the profile, we have to remember to update the local storage value as well, but moreso, if the user changes their profile data from a different device / tab / browser, we won't know that the profile data changed, and we'll be showing the old value to the user
- Also, we don't want to store many sensitive information in localstorage, as it is plain text after all